config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BuilderConstants.java

```diff
@@ -6,11 +6,34 @@
 
 package com.ca.apim.gateway.cagatewayconfig.bundle.builder;
 
+import com.ca.apim.gateway.cagatewayconfig.util.entity.EntityTypes;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Predicate;
+
 @SuppressWarnings("squid:S2068") // sonarcloud believes this is a hardcoded password
 public final class BuilderConstants {
 
+    private BuilderConstants(){
+    }
+
     public static final String STORED_PASSWORD_REF_FORMAT = "${secpass.%s.plaintext}";
 
-    private BuilderConstants(){
+    protected static final Set<String> NON_ENV_ENTITY_TYPES;
+
+    public static final Predicate<Entity> FILTER_ENV_ENTITIES = new Predicate<Entity>() {
+        @Override
+        public boolean test(Entity entity) {
+            return !NON_ENV_ENTITY_TYPES.contains(entity.getType());
+        }
+    };
+
+    static {
+        NON_ENV_ENTITY_TYPES = new HashSet<>();
```

Comment:
instead of static list, use EnvironmentType annotation

Reply:
I don't understand how? EnvironmentType will give me values which I have to check again anyway to find if it is Folder, Encass, Policy or Service.